### PR TITLE
Improve documentation for 0.4 release

### PR DESCRIPTION
**Pull request by wachterjohannes**
This PR improves documentation for 0.4 release

### CODE REVIEW

File: components.rst [hunk 1]
```diff
@@ -0,0 +1,35 @@
+Components
+==========
+The library consists of three main parts.
+
+TaskScheduler
+-------------
+The ``TaskScheduler`` stores tasks and schedules Executions. Foreach tasks there
```
Comment (danrot) [inline components.rst]: `executions`

Comment (danrot) [reply components.rst]: `foreach` is not one english word, must be `for each`

Comment (danrot) [reply components.rst]: And it should be `task` instead of `tasks`

File: components.rst [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+exists always on execution in the status ``planned``, ``running``. If the task
```
Comment (danrot) [inline components.rst]: `exactly one execution`, or should that say something else?

File: components.rst [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+will only one time it only get one execution at all. For recurring tasks the
+execution will always generated for the next run time.
```
Comment (danrot) [inline components.rst]: `For recurring tasks a new execution will be generated when the old one is completed.`?

File: components.rst [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+Each task consists of:
+
+- ``uuid``: unique identifier
+- ``handlerClass``: class-name of the handler which will be executed for this
+  task
+- ``workload``: the workload will be passed to the handler
+- ``intervall``:
+- ``firstExecution``:
+- ``lastExecution``:
```
Comment (danrot) [inline components.rst]: I would write another short explanation or remove the colons.

**Review by danrot — CHANGES_REQUESTED** [review]

**Comment by wachterjohannes** [comment]
@danrot fixed

**Review by danrot — APPROVED** [review]

**Pull request merged**